qt/applications/workbench/workbench/widgets/about/view.py

```diff
@@ -9,18 +9,62 @@
 from qtpy.QtCore import Qt, QSize
 from qtpy.QtGui import QPixmap, QIcon, QGuiApplication, QPainter
 from qtpy.QtWidgets import QVBoxLayout, QHBoxLayout, QSpacerItem, QSizePolicy, QLabel, QDialog, \
-    QGroupBox, QFormLayout, QComboBox, QPushButton, QCheckBox
+    QGroupBox, QFormLayout, QComboBox, QPushButton, QCheckBox, QWidget, QScrollArea
 from mantidqt.widgets import instrumentselector
 
 REFERENCE_HEIGHT = 642
 REFERENCE_WIDTH = 745
 REFERENCE_ASPECT_RATIO = REFERENCE_WIDTH / REFERENCE_HEIGHT
-WIDESCREEN_ASPECT_RATIO = 16/9
+WIDESCREEN_ASPECT_RATIO = 16 / 9
 
 
 class AboutView(QDialog):
-    def __init__(self, parent, presenter, version_text, date_text = None):
+    """
+    Dialog that contains the about widget. If the screen is too small, the dialog will use a scroll area,
+    otherwise it will simply contain the about widget.
+    """
+
+    def __init__(self, parent, presenter, version_text, date_text=None):
         super(AboutView, self).__init__(parent)
+        self.presenter = presenter
+        self.about_widget = AboutViewWidget(self, version_text, date_text)
+
+        layout = QVBoxLayout()
+        layout.setContentsMargins(0, 0, 0, 0)
+        self.setLayout(layout)
+
+        if self.about_widget.is_widget_too_big:
+            self.scroll_area = QScrollArea()
+            self.scroll_area.setWidget(self.about_widget)
+            layout.addWidget(self.scroll_area)
+            self.setMaximumSize(self.about_widget.size())
+
+        else:
+            layout.addWidget(self.about_widget)
+            self.setFixedSize(self.about_widget.size())
+
+    def closeEvent(self, event):
+        self.presenter.save_on_closing()
+        self.deleteLater()
+        super(AboutView, self).closeEvent(event)
+
+    def resizeEvent(self, event):
+        """
+        Hide the scroll bars if the dialog reaches the size of the about widget.
+        """
+        super().resizeEvent(event)
+        if hasattr(self, 'scroll_area'):
+            if self.width() < self.about_widget.width() or self.height() < self.about_widget.height():
+                self.scroll_area.setHorizontalScrollBarPolicy(Qt.ScrollBarAsNeeded)
+                self.scroll_area.setVerticalScrollBarPolicy(Qt.ScrollBarAsNeeded)
+            else:
+                self.scroll_area.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+                self.scroll_area.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
```

Comment:
Can we not just set the scroll area to `ScrollBarAsNeeded` and drop this event handler completely? 
According to the docs it will clear the scroll bars if the content is smaller than the window

Reply:
That was my original intention but it didn't work properly because I've set the maximum size of the widget to the size of the about_widget. Then the problem is that if it's expanded to that maximum size, the scroll bars take up some of the real estate, and thus are required. I could not see any other way of solving that problem.

Reply:
Ah so Qt can't recognise that dropping said scroll bars would actually make the widget fit
It was worth a check, shame the Qt docs don't do what they claim to :man_shrugging: